packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -372,17 +372,28 @@ class FlutterPlugin implements Plugin<Project> {
     }
 
     private Properties getPluginList() {
-        File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
-        Properties allPlugins = readPropertiesIfExist(pluginsFile)
+
         Properties androidPlugins = new Properties()
-        allPlugins.each { name, path ->
-            if (doesSupportAndroidPlatform(path)) {
```

Comment:
@igorakkerman this check should be preventing the issue that you are describing.  I have created a new app and imported `shared_preferences`, I don't see this issue.

Reply:
@blasten Indeed, I could not reproduce the issue with the current `shared_preferences` package, sorry for that. But I do get the message with `location` and `firebase_core`. I created a flutter project where the issue appears: https://github.com/igorakkerman/flutter_57907_example

```
> flutter clean
Deleting build...                                                  319ms
Deleting .dart_tool...                                               7ms
Deleting Generated.xcconfig...                                       2ms
Deleting flutter_export_environment.sh...                            1ms
C:\Lokal\External\flutter_57907_example [master ≡ +0 ~6 -1 !]
> flutter pub get
Running "flutter pub get" in flutter_57907_example...               1,2s
C:\Lokal\External\flutter_57907_example [master ≡ +0 ~7 -0 !]
> flutter run
Using hardware rendering with device Android SDK built for x86 64. If you get graphics artifacts, consider enabling software rendering with
"--enable-software-rendering".
Launching lib\main.dart on Android SDK built for x86 64 in debug mode...
Plugin project :firebase_core_web not found. Please update settings.gradle.
Plugin project :location_web not found. Please update settings.gradle.
Running Gradle task 'assembleDebug'...
Running Gradle task 'assembleDebug'... Done                        38,0s
√ Built build\app\outputs\flutter-apk\app-debug.apk.
```

Reply:
Feel free to reopen this PR again (It got rolled back).  It's good to remove any reference to .flutter-plugins in the plugin.

Reply:
Thanks, @blasten. I re-opened it as #59294 .